File: cpp/09proc/Proc3.cpp
#include <iostream>
#include <cmath>
using namespace std;

void Mean(double X, double Y, double& AMean, double& GMean);

int main()
{
    Task("Proc3");
	double a, b, c, d, am, gm;
	cin >> a >> b >> c >> d;

	Mean(a, b, am, gm);
	cout << am << gm;

	Mean(a, c, am, gm);
	cout << am << gm;

	Mean(a, d, am, gm);
	cout << am << gm;
	
	return 0;
}

void Mean(double X, double Y, double& AMean, double& GMean)
{
	AMean = (Y+X)/2;
	GMean = sqrt(Y*X);
}

File: php/09proc/proc3.php
<?php
function Mean($X, $Y, &$AMean, &$GMean){
        $AMean = ($X + $Y) / 2;
        $GMean = sqrt($X * $Y);
}
$a = mt_rand(1, 50);
$b = mt_rand(1, 50); 
$c = mt_rand(1, 50);
$d = mt_rand(1, 50);
echo "a = ".$a;
echo "<br>b = ".$b;
echo "<br>c = ".$c;
echo "<br>d = ".$d;
Mean($a, $b, $arif, $geom);
$arif = number_format($arif, 2);
$geom = number_format($geom, 2);
echo "<br>Қимати миёнаи арифметики a ва b:  ".$arif;
echo "<br>Қимати миёнаи геометрии a ва b:  ".$geom."<br>";
Mean($a, $c, $arif, $geom);
$arif = number_format($arif, 2);
$geom = number_format($geom, 2);
echo "<br>Қимати миёнаи арифметики a ва c:  ".$arif;
echo "<br>Қимати миёнаи геометрии a ва c:  ".$geom."<br>";
Mean($a, $d, $arif, $geom);
$arif = number_format($arif, 2);
$geom = number_format($geom, 2);
echo "<br>Қимати миёнаи арифметики a ва d:  ".$arif;
echo "<br>Қимати миёнаи геометрии a ва d:  ".$geom;
?>